src/toil/wdl/wdltoil.py

```diff
@@ -624,26 +610,37 @@ def unpack_toil_uri(toil_uri: str) -> Tuple[FileID, str, str]:
 
     return file_id, parent_id, file_basename
 
-def evaluate_output_decls(output_decls: List[WDL.Tree.Decl], all_bindings: WDL.Env.Bindings[WDL.Value.Base], standard_library: ToilWDLStdLibBase) -> WDL.Env.Bindings[WDL.Value.Base]:
+def evaluate_bindings_from_decls(decls: List[WDL.Tree.Decl], all_bindings: WDL.Env.Bindings[WDL.Value.Base], standard_library: ToilWDLStdLibBase,
+                                 include_previous: bool = False, drop_missing_files: bool = False) -> WDL.Env.Bindings[WDL.Value.Base]:
     """
-    Evaluate output decls with a given bindings environment and standard library.
+    Evaluate decls with a given bindings environment and standard library.
     Creates a new bindings object that only contains the bindings from the given decls.
-    Guarantees that each decl in `output_decls` can access the variables defined by the previous ones.
+    Guarantees that each decl in `decls` can access the variables defined by the previous ones.
     :param all_bindings: Environment to use when evaluating decls
-    :param output_decls: Decls to evaluate
+    :param decls: Decls to evaluate
     :param standard_library: Standard library
-    :return: New bindings object with only the output_decls
-    """
-    # all_bindings contains output + previous bindings so that the output can reference its own declarations
-    # output_bindings only contains the output bindings themselves so that bindings from sections such as the input aren't included
-    output_bindings: WDL.Env.Bindings[WDL.Value.Base] = WDL.Env.Bindings()
-    for output_decl in output_decls:
-        output_value = evaluate_decl(output_decl, all_bindings, standard_library)
-        drop_if_missing_with_workdir = partial(drop_if_missing, work_dir=standard_library.execution_dir)
-        output_value = map_over_typed_files_in_value(output_value, drop_if_missing_with_workdir)
-        all_bindings = all_bindings.bind(output_decl.name, output_value)
-        output_bindings = output_bindings.bind(output_decl.name, output_value)
-    return output_bindings
+    :param include_previous: Whether to include the existing environment in the new returned environment. This will be false for outputs where only defined decls should be included
+    :param drop_missing_files: Whether to coerce nonexistent files to null. The coerced elements will be checked that the transformation is valid.
+    Currently should only be enabled in output sections, see https://github.com/openwdl/wdl/issues/673#issuecomment-2248828116
+    :return: New bindings object with only the decls
```

Comment:
Unless `include_previous` is true in which case the existing bindings are also carried through, right?